Tests/SonarScanner.MSBuild.PreProcessor.Test/SonarCloudWebServerTest.cs

```diff
@@ -267,21 +267,85 @@ public async Task DownloadCache_CacheHit(string tokenKey)
             handler.VerifyAll();
         }
 
+        [TestMethod]
+        public async Task DownloadCache_PrepareRead_UnsuccessfulResponse()
+        {
+            const string cacheBaseUrl = "https://www.cacheBaseUrl.com";
+            var cacheFullUrl = $"https://www.cacheBaseUrl.com/v1/sensor_cache/prepare_read?organization={Organization}&project=project-key&branch=project-branch";
+            var handler = MockHttpHandler(cacheFullUrl, "irrelevant", HttpStatusCode.Forbidden);
+            sut = new SonarCloudWebServer(MockIDownloader(cacheBaseUrl), version, logger, Organization, handler.Object);
+            var localSettings = CreateLocalSettings(ProjectKey, ProjectBranch, Organization, Token);
+
+            var result = await sut.DownloadCache(localSettings);
+
+            result.Should().BeEmpty();
+            logger.AssertSingleDebugMessageExists("Incremental PR analysis: an error occurred while retrieving the cache entries! 'prepare_read' did not respond successfully.");
+            handler.VerifyAll();
+        }
+
+        [TestMethod]
+        public async Task DownloadCache_PrepareRead_EmptyResponse()
+        {
+            const string cacheBaseUrl = "https://www.cacheBaseUrl.com";
+            var cacheFullUrl = $"https://www.cacheBaseUrl.com/v1/sensor_cache/prepare_read?organization={Organization}&project=project-key&branch=project-branch";
+            var handler = MockHttpHandler(cacheFullUrl, string.Empty);
+            sut = new SonarCloudWebServer(MockIDownloader(cacheBaseUrl), version, logger, Organization, handler.Object);
+            var localSettings = CreateLocalSettings(ProjectKey, ProjectBranch, Organization, Token);
+
+            var result = await sut.DownloadCache(localSettings);
+
+            result.Should().BeEmpty();
+            logger.AssertSingleDebugMessageExists("Incremental PR analysis: an error occurred while retrieving the cache entries! 'prepare_read' response was empty.");
+            handler.VerifyAll();
+        }
+
+        [TestMethod]
+        public async Task DownloadCache_PrepareRead_CacheDisabled()
+        {
+            const string cacheBaseUrl = "https://www.cacheBaseUrl.com";
+            var cacheFullUrl = $"https://www.cacheBaseUrl.com/v1/sensor_cache/prepare_read?organization={Organization}&project=project-key&branch=project-branch";
+            var handler = MockHttpHandler(cacheFullUrl, $@"{{ ""enabled"": ""false"", ""url"":""https://www.irrelevant.com"" }}");
```

Comment:
unfortunately `https://www.irrelevant.com` already exists. can we use maybe our own website or some mock data?